drivers/ethernet/eth_enc28j60.c

```diff
@@ -757,6 +757,39 @@ static enum ethernet_hw_caps eth_enc28j60_get_capabilities(const struct device *
 		;
 }
 
+static int eth_enc28j60_set_config(const struct device *dev,
+				   enum ethernet_config_type type,
+				   const struct ethernet_config *config)
+{
+	struct eth_enc28j60_runtime *context = dev->data;
+
+	/* Compile time check that the memcpy below won't overflow */
+	BUILD_ASSERT(sizeof(context->mac_address) <= sizeof(config->mac_address.addr),
+		     "ENC28j60 Runtime MAC address buffer too small");
+
+	if (type == ETHERNET_CONFIG_TYPE_MAC_ADDRESS) {
+		memcpy(context->mac_address, config->mac_address.addr,
+		       sizeof(config->mac_address.addr));
+		eth_enc28j60_init_mac(dev);
+
+		if (context->iface) {
```

Comment:
Use explicit pointer checks

```suggestion
		if (context->iface != NULL) {
```